test/test-logger.js

```diff
@@ -103,14 +112,66 @@ describe('logger', function () {
             logger.close();
         });
 
+        it('sends compressed log as an object with extra fields', function (done) {
+            var logger = createLogger({
+                bufferSize: 1,
+                callback: onDone,
+                extraFields: {
+                    extraField1: 'val1',
+                    extraField2: 'val2'
+                },
+                doCompress: true
+            });
+
+            sinon.spy(logger, '_tryToSend');
+            var logMsg = {
+                message: 'hello there from test'
+            };
+
+            logger.log(logMsg);
+
+            function onDone() {
+                assert(logger._tryToSend.getCall(0).args[0].headers['content-encoding'] == 'gzip');
+                var unzipBody = JSON.parse(zlib.gunzipSync(logger._tryToSend.getCall(0).args[0].body));
+                assert(unzipBody.message == logMsg.message);
+                assert(unzipBody.extraField1 == 'val1');
```

Comment:
These 2 lines are making a duplication of the value.
Use the `logger` object to read the value you want to check.